### DISCUSSION TITLE
extending howfairis

### DISCUSSION THREAD

**Discussion by alee** [fair-software/howfairis#338] [General] [answered]
We're starting to look into automated FAIR assessments of models in the [comses model library](https://www.comses.net/codebases/) and are very glad this project and the accompanying github action exist, kudos for all the good work! 

I think this is likely related to https://github.com/fair-software/howfairis/issues/334 but do you have any recommendations / guidance on adding new compliance checks? For example, we might want to perform some cursory check for detailed narrative documentation that follows the [ODD protocol](https://www.jasss.org/23/2/7.html) or a check for containerization / Open Science Grid compliant Dockerfile.

**Comment by jspaaks** [comment, marked as answer]
Hi @alee,

sorry this went unanswered for such a long time. As to your question, extending the list of checks is currently not very convenient (indeed, hence the issue #334) but basically, it would entail extending one of the five mixin classes. For example, if you wanted to add a check for an additional registry, you would go to the corresponding mixin ([here](https://github.com/fair-software/howfairis/blob/0.14.1/howfairis/mixins/registry_mixin.py#L6)), add a method to it, for example `def has_my_repo_badge(self)`, and call it from this list:

https://github.com/fair-software/howfairis/blob/0.14.1/howfairis/mixins/registry_mixin.py#L15-L24

While this works from a technical prespective, I anticipate some problems with how the results are presented, because the two checks you mention are a bit of a square peg in the round holes that https://fair-software.nl provides.

It may be worthwhile to keep an eye on #334, since that will probably make implementing arbitrary checks easier. Sorry I can't make any concrete promises, we're in the middle of reorganizing some of our decision making regarding budgets which directly affect whether I will be able to work on this.
